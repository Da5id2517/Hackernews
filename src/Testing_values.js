export const list = [
  {
    title: 'Naslov',
    url: 'https://lemonparty.org',
    author: 'SomeBloke',
    num_comment: 3,
    points: 4,
    objectID: 0
  },
  {
    title: 'Seheron',
    url: 'https://youtube.org',
    author: 'Tseries',
    num_comment: 12,
    points: 3,
    objectID: 1
  }
]

export const blank_func = () => {};
export const test_string = 'test_string';
