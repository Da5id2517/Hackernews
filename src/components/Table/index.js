import React, {Component} from 'react';
import Sort from '../Sort';
import Button from '../Button';
import PropTypes from 'prop-types';
import {
  smallColumn,
  mediumColumn,
  largeColumn,
  SORTS
} from '../../constants';


export default class Table extends Component {
  constructor(props) {
    super(props);
    this.state = {
      sortKey: 'NONE',
      isSortReversed: false,
    }

    this.onSort = this.onSort.bind(this);
  };

  onSort(sortKey) {
    const isSortReversed = sortKey === this.state.sortKey && !this.state.isSortReversed
    this.setState({isSortReversed: isSortReversed, sortKey: sortKey})
  }

  render() {
  const {list, onDismiss} = this.props;
  const {sortKey, isSortReversed} = this.state;
  const sortedList = SORTS[sortKey](list);
  const reverseSortedList = isSortReversed ?
    sortedList.reverse() :
    sortedList;
  return (
    <div className='table'>
      <div className='table-header'>
        <span style={{width: largeColumn}}>
          <Sort
            sortKey={'TITLE'}
            onSort={this.onSort}
            activeSortKey={sortKey}>
            Title
          </Sort>
        </span>
        <span style={{width: mediumColumn}}>
          <Sort
            sortKey={'AUTHOR'}
            onSort={this.onSort}
            activeSortKey={sortKey}>
            Author
          </Sort>
        </span>
        <span style={{width:smallColumn}}>
          <Sort
            sortKey={'COMMENTS'}
            onSort={this.onSort}
            activeSortKey={sortKey}>
            Comments
          </Sort>
        </span>
        <span style={{width:smallColumn}}>
          <Sort
            sortKey={'POINTS'}
            onSort={this.onSort}
            activeSortKey={sortKey}>
            Points
          </Sort>
        </span>
        <span style={{width:smallColumn}}>
          Archive
        </span>
      </div>
      {reverseSortedList.map(item =>
        <div key='obejctID' className='table-row'>
          <span style={largeColumn}>
            <a href={item.url}> {item.title} </a>
            </span>
            <span style={mediumColumn}> {item.author} </span>
            <span style={smallColumn}> {item.num_comment} </span>
            <span style={smallColumn}> {item.points} </span>
            <span style={smallColumn}>
            <Button
            onClick={() => onDismiss(item.objectID)}
            className='button-inline'>
              Dismiss
            </Button>
            </span>
          </div>
        )
      }
      </div>
    );
  }
}

Table.propTypes = {
  list: PropTypes.arrayOf(
    PropTypes.shape({
      objectID: PropTypes.string.isrequired,
      author: PropTypes.string,
      url: PropTypes.string,
      num_comment: PropTypes.number,
      points: PropTypes.number,
    }).isRequired
  ),
  onDismiss: PropTypes.func.isRequired,
};
