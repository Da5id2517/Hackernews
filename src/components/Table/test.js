import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Table from './';
import {list, blank_func} from '../../Testing_values.js';


Enzyme.configure({adapter: new Adapter});

describe('Table', () => {
  it('renders', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Table
      list={list}
      onDismiss={blank_func}
      />,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  })

  it('has valid snappy', () => {
    const comp = renderer.create(
      <Table
      list={list}
      onDismiss={blank_func}
      />
    );
    let tree = comp.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('lenght of props', () => {
    const element = shallow(
      <Table
      list={list}
      onDismiss={blank_func}
      />
    );
    expect(element.find('.table-row').length).toBe(2);
  })
})
