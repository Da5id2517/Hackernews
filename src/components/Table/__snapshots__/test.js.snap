// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table has valid snappy 1`] = `
<div
  className="table"
>
  <div
    className="table-header"
  >
    <span
      style={
        Object {
          "width": Object {
            "width": "40%",
          },
        }
      }
    >
      <button
        className="button-inline"
        onClick={[Function]}
        type="button"
      >
        Title
      </button>
    </span>
    <span
      style={
        Object {
          "width": Object {
            "width": "30%",
          },
        }
      }
    >
      <button
        className="button-inline"
        onClick={[Function]}
        type="button"
      >
        Author
      </button>
    </span>
    <span
      style={
        Object {
          "width": Object {
            "width": "10%",
          },
        }
      }
    >
      <button
        className="button-inline"
        onClick={[Function]}
        type="button"
      >
        Comments
      </button>
    </span>
    <span
      style={
        Object {
          "width": Object {
            "width": "10%",
          },
        }
      }
    >
      <button
        className="button-inline"
        onClick={[Function]}
        type="button"
      >
        Points
      </button>
    </span>
    <span
      style={
        Object {
          "width": Object {
            "width": "10%",
          },
        }
      }
    >
      Archive
    </span>
  </div>
  <div
    className="table-row"
  >
    <span
      style={
        Object {
          "width": "40%",
        }
      }
    >
      <a
        href="https://lemonparty.org"
      >
         
        Naslov
         
      </a>
    </span>
    <span
      style={
        Object {
          "width": "30%",
        }
      }
    >
       
      SomeBloke
       
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
       
      3
       
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
       
      4
       
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      <button
        className="button-inline"
        onClick={[Function]}
        type="button"
      >
        Dismiss
      </button>
    </span>
  </div>
  <div
    className="table-row"
  >
    <span
      style={
        Object {
          "width": "40%",
        }
      }
    >
      <a
        href="https://youtube.org"
      >
         
        Seheron
         
      </a>
    </span>
    <span
      style={
        Object {
          "width": "30%",
        }
      }
    >
       
      Tseries
       
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
       
      12
       
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
       
      3
       
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      <button
        className="button-inline"
        onClick={[Function]}
        type="button"
      >
        Dismiss
      </button>
    </span>
  </div>
</div>
`;
